crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -0,0 +1,72 @@
+use {
+    super::{auction, eth},
+    crate::infra,
+    anyhow::Result,
+};
+
+pub mod encoded;
+pub mod fees;
+pub mod observation;
+pub mod surplus;
+pub mod transaction;
+
+pub use {
+    encoded::Encoded,
+    fees::Fees,
+    surplus::{NormalizedSurplus, Surplus},
+    transaction::Transaction,
+};
+
+/// A transaction that settles a settlement. Interacts with the settlement
+/// contract `settle` function.
+pub struct Settlement {
+    encoded: Encoded,
+    transaction: transaction::Transaction,
+    receipt: transaction::Receipt,
+}
+
+impl Settlement {
+    pub async fn new(tx: eth::TxId, eth: infra::Ethereum) -> Result<Self, Error> {
+        let transaction = eth
+            .transaction(tx)
+            .await?
+            .ok_or(Error::TransactionNotFound)?;
+        let receipt = eth
+            .transaction_receipt(tx)
+            .await?
+            .ok_or(Error::TransactionNotFound)?;
+        let domain_separator = eth.contracts().settlement_domain_separator();
+        let encoded = Encoded::new(transaction.input(), domain_separator)?;
+        Ok(Self {
+            encoded,
+            transaction,
+            receipt,
+        })
+    }
+
+    pub fn auction_id(&self) -> auction::Id {
+        self.encoded.auction_id()
+    }
+
+    pub fn transaction(&self) -> &transaction::Transaction {
+        &self.transaction
+    }
+
+    pub fn transaction_receipt(&self) -> &transaction::Receipt {
+        &self.receipt
+    }
+
+    pub fn encoded(&self) -> &Encoded {
+        &self.encoded
+    }
```

Comment:
Do we need to leak the whole encoded settlement here (which transitively leaks all the structs that are defined in this PR)? I think we would have to worry much less about how to factor things if we can keep the domain model small and explicitly provide the functionality the domain needs.

Reply:
No, we don't have to. I just anticipated that we will have a lot of functionalities to be done using the `settlement`. But we can also keep everything private, and have an extension impl in modules surplus.rs, fees.rs, scoring.rs etc. I am fine with both.

Reply:
removed this function